cdap-ui/graphql/resolvers/applicationResolvers.js

```diff
@@ -27,34 +27,41 @@ const applicationsResolver = {
   Query: {
     applications: async (parent, args, context) => {
       const namespace = args.namespace;
+      const artifactName = args.artifaceName;
       const options = resolversCommon.getGETRequestOptions();
-      options.url = urlHelper.constructUrl(cdapConfig, `/v3/namespaces/${namespace}/apps`);
+      options.url = urlHelper.constructUrl(
+        cdapConfig,
+        `/v3/namespaces/${namespace}/apps?artifactName=${artifactName}`
```

Comment:
check for the existence of artifactName first before appending this query parameter. If it doesn't exist, don't append

Reply:
The graphql schema is defined as `applications(namespace: String = "default", artifactName: String = "cdap-data-pipeline,cdap-data-streams"): [ApplicationRecord]!`, which passes a default value for `artifactName` if one is not specified in the query. Hence, based on how the schema is defined, there will always be a value for `artifactName` in this resolver.

Reply:
don't think we should do that. I think the argument should be passed by whoever is using it. Because that means I can't get a list of all applications (even though I know it won't work at the moment because of limited programs resolvers)

Reply:
Done